packages/api/src/controllers/stream.ts

```diff
@@ -447,9 +450,6 @@ export async function getRecordingFields(
   const isUnused = !session.lastSeen && session.createdAt < readyThreshold;
 
   const recordingStatus = isReady ? "ready" : isUnused ? "none" : "waiting";
```

Comment:
I think we should also change this to be more compatible with recording v2. This logic works fine for v1, where the recording readiness was loosely based on when the stream ended (`lastSeen`), but I think we have more accurate information here for recording v2.

So WDYT of fixing this status as well? You can use the same `session.version === "v2"` to tell if we should use the asset state for this, and/or maybe even move this logic to `getRecordingUrls`, or implement it based on the response from `getRecordingUrls`. 

Can also consider having a new `recordingStatus` value here for source playback like `"source"`. But otherwise it should be fine to keep it as `waiting` anyway, just like the asset that gets a `playbackUrl` but stays in `processing`.

Reply:
Ah nice I was wondering what that time based logic was about :)

Reply:
I've done a little refactor which I think makes things much simpler, see what you think

Reply:
Neat refact, I think it's simpler to follow the logic now with v1/v2 totally separate